libbeat/processors/add_kubernetes_metadata/config.go

```diff
@@ -43,11 +48,33 @@ type Enabled struct {
 
 type PluginConfig []map[string]common.Config
 
+func getSystemKubeConfig() string {
+	envKubeConfig := os.Getenv("KUBECONFIG")
+	if _, err := os.Stat(envKubeConfig); !os.IsNotExist(err) {
+		return envKubeConfig
+	}
+	homeKubeConfig := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	if _, err := os.Stat(homeKubeConfig); !os.IsNotExist(err) {
+		return homeKubeConfig
+	}
+	return ""
+}
+
+func inCluster() bool {
+	_, err := rest.InClusterConfig()
+	if err == nil {
+		return true
+	}
+	return false
+}
```

Comment:
Actually, the field `InCluster` can be removed.

Regarding the searching of environmental variables or `kubeconfig` in $HOME directory: 
If we are not in an "incluster" mode then the autodetection will just fail if users have not set `kube_config` in their configuration cause it will not be able to create a client etc. So auto-detection will be actually available only in "Incluster" deployments or we should require from the user to set the `kubeconfig`.

In addition this code has do only with `add_kubernetes_metadata` processor, so is it possible to affect other features too (really not sure about that, just asking).

So what we should do? I'm open on any proposal :)

Reply:
> Actually, the field `InCluster` can be removed.
> 
:+1: 

> Regarding the searching of environmental variables or `kubeconfig` in $HOME directory:
> If we are not in an "incluster" mode then the autodetection will just fail if users have not set `kube_config` in their configuration cause it will not be able to create a client etc. So auto-detection will be actually available only in "Incluster" deployments or we should require from the user to set the `kubeconfig`.
> 
I think that failing if in-cluster configuration is not available and kubeconfig is not set would be perfectly fine, and it would be consistent with the rest of kubernetes features.

Said that, I see that needing to set kubeconfig in all Kubernetes features can be cumbersome, we could try to explore the use of `KUBECONFIG` env variable as an alternative to set it everywhere. But if something needs to be done for that I would do it in the common helpers, and in a separate PR.

> In addition this code has do only with `add_kubernetes_metadata` processor, so is it possible to affect other features too (really not sure about that, just asking).
> 
> So what we should do? I'm open on any proposal :)

I'd say to:
* Keep it failing if in-cluster configuration is not available and kubeconfig is not set. In any case the most common case for Kubernetes is to have in-cluster configuration working.
* Consider leveraging `KUBECONFIG` for all our Kubernetes features, but as a separate issue/PR.

Reply:
👍